sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -49,50 +48,58 @@
 class SetStatementAnalyzer {
 
     private static final Logger LOGGER = LogManager.getLogger(SetStatementAnalyzer.class);
+    private final Functions functions;
 
-    public static AnalyzedStatement analyze(SetStatement node) {
+    public SetStatementAnalyzer(Functions functions) {
+        this.functions = functions;
+    }
+
+    public AnalyzedStatement analyze(SetStatement<Expression> node,
+                                     ParamTypeHints typeHints,
+                                     CoordinatorTxnCtx txnCtx) {
 
         boolean isPersistent = node.settingType().equals(SetStatement.SettingType.PERSISTENT);
-        Map<String, List<Expression>> settings = new HashMap<>();
-        Assignment<Expression> assignment;
+        Map<String, List<Symbol>> settings = new HashMap<>();
+
+        var exprAnalyzer = new ExpressionAnalyzer(
+            functions,
+            txnCtx,
+            typeHints,
+            FieldProvider.FIELDS_AS_LITERAL,
```

Comment:
I think field usages inside `SET` statements are not supported.

Reply:
It fails when I set it to `FieldProvider.UNSUPPORTED`, what is the exact purpose of this ?

Reply:
What does fail? Using a column ident/reference/field inside a setting key or value isn't supported, so the ``UNSUPPORTED`` FieldProvider should be used as this will raise otherwise.

Reply:
e.g. Running https://github.com/crate/crate/blob/3ff3a9eb2eeb86c44158db2dc47931dc2e133084/sql/src/test/java/io/crate/planner/PlannerTest.java#L31 with `FieldProvider.UNSUPPORTED` throws a 
`java.lang.UnsupportedOperationException: Columns cannot be used in this context. Maybe you wanted to use a string literal which requires single quotes: 'stats.jobs_log_size'`

Reply:
Ah ok,  so keys are treated as idents by the parser cause assignments are used here.
We could follow up later here to use `GenericProperties` for set stmts maybe.